### ISSUE TITLE
Ability to turn off the 2D Views Ruller guide permanently in Editor Settings

### ISSUE THREAD

**Issue by Bat-Computer** [godotengine/godot-proposals#8795]
### Describe the project you are working on

This is for 2D projects user experience option

### Describe the problem or limitation you are having in your project

There is no way to permanently turn off the Ruller guide in the 2D view. In other software you can toggle it on or off permanently, for example Photoshop. 

I find it clutters the workspace and rarely have use for it.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I find it clutters the workspace and rarely have use for it as I'm sure many other users would say.

It would still be nice to have the option within the 2D view to toggle it on and off where necassary, but also a permanent one too.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I'm not sure the ideal location in the Editor Settings to place it, perhaps 'Editor' or 'Theme'?

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I would expect it to be straight-forward to change.

### Is there a reason why this should be core and not an add-on in the asset library?

It's a setting location change ;)

**Comment by AThousandShips** [comment]
You can? It's in the `View -> Show Rulers` menu in the 2D editor

**Comment by AThousandShips** [comment]
I'm not sure it's that annoying to have to turn it off once in every project, and you can't really have it be half-on so to say, how would you handle that in the settings? How would you have it be both togglable *and* a setting?

**Comment by Bat-Computer** [comment]
Yes I understand that there is a toggle already in the 2D view as I stated ;) but it is not saved perminently between **new** projects. For example in Photoshop you can have rulers on or off, when you restart Photoshop with a new project it is saved in Settings if you set it that way. If you wish to toggle Rulers on/off, you still can in the menu just like Godot does.

**Comment by AThousandShips** [comment]
And how would that be handled? When you have it off in the settings and want it on in the editor? That's important information here 🙂 This is a setting that makes more sense to be in the project, not in the editor settings IMO

**Comment by KoBeWi** [comment]
Currently the ruler visibility is not even a project setting, it's just editor layout value. It's stored per-scene, like grid settings. Editor setting could determine the default value when opening a new project, but we don't have other setting like that and IMO it would just clutter the setting list. Not really worth it for something that you toggle once per project.

**Comment by Bat-Computer** [comment]
Some users don't want clutter, same for me. The rullers could be switch on/off be default in the Editor Settings. This would enable the user to enable disable on the fly if they wanted the rulers, but when starting a new project Godot would remember the Editor Settings over the menu option toggle.

For me it's the first thing I do when I start a new 2D project, that and change the Default Clear Color to a darker hue.

I didn't think it would be an unreasonable request, but it seems it's a lot more work that I imagined it would be. Thanks anyway.

**Comment by KoBeWi** [comment]
> it seems it's a lot more work that I imagined it would be

In terms of the work involved this change is actually trivial.
